test/parallel/test-http-request-invalid-method-error.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const http = require('http');
+
+assert.throws(
+  () => { http.request({method: '\0'}); },
+  (error) => {
+    if ((error instanceof TypeError)
+       && /Method must be a valid HTTP token/.test(error)) {
+      return true;
```

Comment:
Instead of returning `true` inside the `if`, you can just return the result of the conditional checks.